File: Seccion 11 - Funciones en C++/Seccion_11_Dart/12_ejercicio_10.dart

import 'dart:io';

class Fecha {
  int anio, mes, dia;

  Fecha();
}

void calc_anios(int totalDias, Fecha fecha) {
  fecha.anio = (totalDias / 365).floor();
  totalDias %= 365;
  fecha.mes = (totalDias / 30).floor();
  fecha.dia = totalDias % 30;
}

void main() {

  int totalDias;

  stdout.writeln('Digite el número total de días: ');
  totalDias = int.parse(stdin.readLineSync());

  Fecha fecha = new Fecha();

  calc_anios(totalDias, fecha);

  print('Fecha actual: ${fecha.dia + 1}/${fecha.mes + 1}/${fecha.anio + 2000}');


}

File: Seccion 11 - Funciones en C++/Seccion_11_Python/12_ejercicio_10.py

anio = 0
mes = 0
dia = 0

def calc_anio(total_dias):
    global anio, mes, dia
    
    anio = int(total_dias / 365)
    total_dias %= 365
    mes = int(total_dias / 30)
    dia = total_dias % 30

total_dias = int(input('Digite el número total de días: '))

calc_anio(total_dias)

print('Fecha actual: {}/{}/{}'.format(dia + 1, mes + 1, anio + 2000))